test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/ZookeeperServiceOnK8.java

```diff
@@ -0,0 +1,244 @@
+import java.util.stream.Collectors;
+
+import static io.pravega.test.system.framework.TestFrameworkException.Type.RequestFailed;
+
+/**
+ * Manage Zookeeper service on k8 cluster.
+ */
+@Slf4j
+public class ZookeeperServiceOnK8 implements Service {
+
+    private static final String NAMESPACE = "default";
+    private static final Integer MIN_READY_SECONDS = 10; // minimum duration the operator is up and running to be considered ready.
+    private static final String CUSTOM_RESOURCE_GROUP = "zookeeper.pravega.io";
+    private static final String CUSTOM_RESOURCE_VERSION = "v1beta1";
+    private static final String CUSTOM_RESOURCE_PLURAL = "zookeeper-clusters";
+    private static final String CUSTOM_RESOURCE_KIND = "ZookeeperCluster";
+    private static final String ID = "zk";
+    private static final String OPERATOR_ID = "zookeeper-operator";
+    private static final int ZKPORT = 2181;
+    private static final String TCP = "tcp://";
+    private final K8Client k8Client;
+
+    public ZookeeperServiceOnK8() {
+        k8Client = new K8Client();
+    }
+
+    @Override
+    public void start(boolean wait) {
+        Futures.getAndHandleExceptions(k8Client.createCRD(getZKOperatorCRD())
+                                               .thenCompose(v -> k8Client.createClusterRole(getClusterRole()))
+                                               .thenCompose(v -> k8Client.createClusterRoleBinding(getClusterRoleBinding()))
+                                               //deploy zk operator.
+                                               .thenCompose(v -> k8Client.createDeployment(NAMESPACE, getDeployment()))
+                                               // wait until zk operator is running, only one instance of operator is running.
+                                               .thenCompose(v -> k8Client.waitUntilPodIsRunning(NAMESPACE, "name", OPERATOR_ID, 1))
+                                               // request operator to deploy zookeeper nodes.
+                                               .thenCompose(v -> k8Client.createAndUpdateCustomObject(CUSTOM_RESOURCE_GROUP, CUSTOM_RESOURCE_VERSION, NAMESPACE, CUSTOM_RESOURCE_PLURAL, getZookeeperDeployment(getID(), 1))), t -> new TestFrameworkException(RequestFailed, "Failed to deploy zookeeper operator/service", t));
+        if (wait) {
+            Futures.getAndHandleExceptions(k8Client.waitUntilPodIsRunning(NAMESPACE, "app", ID, 1),
+                                           t -> new TestFrameworkException(RequestFailed, "Failed to deploy zookeeper service", t));
+        }
+    }
+
+
+    @Override
+    public void stop() {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public void clean() {
+    }
+
+    @Override
+    public String getID() {
+        return ID;
+    }
+
+    @Override
+    public boolean isRunning() {
+        // return false since mutliple invocations of start does not have any side effects.
+        return false;
```

Comment:
Can you elaborate a bit more why this has not been properly implemented?

Reply:
implemented it